packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx

```diff
@@ -114,15 +115,27 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
             <span
               className={ css('ms-DatePicker-nextMonth js-nextMonth', styles.nextMonth) }
               onClick={ this._onSelectNextMonth }
-              onKeyDown={ this._onKeyDown.bind(this, this._onSelectNextMonth) }
+              onKeyDown={ this._onSelectNextMonth }
               aria-controls={ dayPickerId }
               aria-label={ strings.nextMonthAriaLabel }
               role='button'
               tabIndex={ 0 }>
               <Icon iconName={ getRTL() ? leftNavigationIcon : rightNavigationIcon } />
             </span >
           </div >
-          <div className={ css('ms-DatePicker-headerToggleView js-showMonthPicker', styles.headerToggleView) } />
+          {
+            this.props.onHeaderClick ?
+              <div
+                className={ css('ms-DatePicker-headerToggleView js-showMonthPicker', styles.headerToggleView) }
+                onClick={ () => this._onHeaderClick(false) }
+                onKeyDown={ () => this._onHeaderClick(true) }
```

Comment:
Why are these passing different focus values? Also, avoid using the lambdas here, create separate functions for these if needed.